### PR TITLE
Updating this BSP

### PR DESCRIPTION
**Pull request by romancardenas**
I bought a Rust board and wanted to use this BSP. However, it appeared not to be updated and I wasn't able to compile it.
This new version uses the latest (I believe) versions of the corresponding HALs and now uses the esp-println crate to print messages via the JTAG interface. I also marked shared-bus as optional (it only applies to the sensors feature).
Additionally, I added supplementary files to debug the applications using VSCode and Cortex-Debug. I hope this can help as guideline to do better debugging.

When debugging in VSCode, I found the following limitations:
- You must first flash the example you want to debug and then start the debugging process. Currently, Cortex-Debug cannot be used to flash directly an application.
- I wasn't able to see the log messages while debugging using VSCode.

Cheers!

### CODE REVIEW

**Comment by romancardenas** [comment]
I forgot to mention: now, all the dependencies point to crates.io . I think this approach is better than using specific commits from GitHub. I would also like this crate being published on crates.io.

File: Cargo.toml [hunk 1]
```diff
@@ -4,27 +4,16 @@ version = "0.1.0"
...
-rev      = "9eb1cb4"
-features = ["esp32c3", "exception-handler", "panic-handler", "print-jtag-serial"]
-
-[dependencies.icm42670]
-git      = "https://github.com/jessebraham/icm42670/"
-rev      = "eb0483b"
-optional = true
+esp-backtrace = { version = "0.2.0", features = ["esp32c3", "exception-handler", "panic-handler", "print-jtag-serial"] }
+esp-println   = { version = "0.3.0", features = ["esp32c3"] }
+esp32c3-hal   = "0.1.0"
```
Comment (jessebraham) [inline Cargo.toml:9]:
```suggestion
esp32c3-hal   = "0.2.0"
```

I've just today published a new version of this package (there were some issues with the previous version), so would you mind updating this please?

**Comment by jessebraham** [comment]
Sorry for taking so long to get to this, it got buried in my notifications. As you can see I've been neglecting this package, so thank you for getting things back up to date!

I just left the one comment and otherwise I think it looks good. If you would be able to add some information to the README regarding debugging that would be great, but if you don't feel like it that's totally fine of course and I can do it at some point.

**Comment by romancardenas** [comment]
Great! Thanks for your comments. I'll write a brief how-to in the README file this week and update the PR.

**Comment by romancardenas** [comment]
I pushed a new commit with the changes you requested. So far, I wasn't able to configure the launch profile for seeing the logs while debugging with cortex-debug. However, I found an extra plugin that allows us to open a serial monitor while debugging. This serves as a workaround. I'll ask the guys of cortex-debug for help, I'm sure we can configure their plugin properly and skip this extra dependency.

**Review by jessebraham — APPROVED** [review]
Thank you again for your contributions, this is looking good!

**Pull request merged**